pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -129,6 +130,19 @@ func (rc *reconciler) syncStates() {
 	}
 }
 
+func (rc *reconciler) isExclusiveVolume(volumeSpec *volume.Spec) bool {
+	if volumeSpec.PersistentVolume == nil {
+		return false
+	}
+	// check if this volume can be attached to multiple PODs/nodes, if yes, return false
+	for _, accessMode := range volumeSpec.PersistentVolume.Spec.AccessModes {
```

Comment:
Ok, now I understand your point. True, we can not use access mode from the PV. I did not take into account that volumes are also possible without a PV. But using the readOnly field is also not a viable option here as it does not tell us if attaching to multiple nodes is OK or not. A GCE persistent disk for example can never be attached to 2 nodes at the same time, no matter if readOnly or not. Even if it's possible to attach it to 2 pods, it's only possible on the same node (if k8s even allows this, not sure).

In case the PV is nil, we should probably make a decision based on the volume/disk type. For example, GCE/AWS/Azure disks are not allowed to attach to multiple nodes. Network based volumes like NFS are allowed to attach to multiple nodes. And so on...